core/src/main/java/hudson/model/Job.java

```diff
@@ -611,15 +612,10 @@ public Collection<?> getOverrides() {
         return r;
     }
 
-    public List<Widget> getWidgets() {
-        ArrayList<Widget> r = new ArrayList<>();
-        r.add(createHistoryWidget());
-        return r;
-    }
-
     /**
-     * @see LazyBuildMixIn#createHistoryWidget
+     * @deprecated see {@link LazyBuildMixIn#createHistoryWidget()}
      */
+    @Deprecated(forRemoval = true, since = "TODO")
```

Comment:
If this is deprecated for removal, then why is `AbstractProject#createHistoryWidget` still calling it?

Reply:
As is `WorkflowJob`. There is a reference to another method but it is unclear from the Javadoc how the deprecation is supposed to be resolved. By deleting the override?

Reply:
All `Widget` instances are now instantiated via `WidgetFactory` extension point. Overrides should be deleted.

Filed https://github.com/jenkinsci/jenkins/pull/9268 to properly clean this up.